crypto/ec/ecx_meth.c

```diff
@@ -1154,19 +1187,19 @@ static int s390x_pkey_ecd_keygen25519(EVP_PKEY_CTX *ctx, EVP_PKEY *pkey)
         0x66, 0x66, 0x66, 0x66, 0x66, 0x66, 0x66, 0x66,
     };
     unsigned char x_dst[32], buff[SHA512_DIGEST_LENGTH];
-    ECX_KEY *key;
+    ECX_KEY *key = NULL;
     unsigned char *privkey = NULL, *pubkey;
```

Comment:
Unless I am mistaken these 3 functions below are now identical - except for the id, and the ECErr function names.
So it could be refactored to be a single function,

Reply:
Not quite although very similar. There are differences in digests, field multiplication function, key lengths. It could possibly still be refactored - but not in this PR.